File: frontend/src/types/portfolio.ts
export interface Position {
  id: string;
  market_id: string;
  market_question: string;
  market_category: string | null;
  side: string;
  shares: number;
  avg_price: number;
  current_price: number | null;
  current_value: number | null;
  unrealized_pnl: number | null;
  cost_basis: number;
  status: string;
  opened_at: string;
}

export interface PortfolioSummary {
  id: string;
  balance: number;
  total_deposited: number;
  total_pnl: number;
  total_trades: number;
  winning_trades: number;
  losing_trades: number;
  win_rate: number;
  open_positions: number;
  today_pnl: number;
  today_trades: number;
  daily_spend_used: number;
  daily_spend_limit: number;
}

File: backend/app/schemas/portfolio.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class PortfolioResponse(BaseModel):
    id: uuid.UUID
    balance: Decimal
    total_deposited: Decimal
    total_pnl: Decimal
    total_trades: int
    winning_trades: int
    losing_trades: int
    win_rate: float
    open_positions: int
    today_pnl: Decimal
    today_trades: int
    daily_spend_used: Decimal
    daily_spend_limit: Decimal

    model_config = {"from_attributes": True}


class PositionResponse(BaseModel):
    id: uuid.UUID
    market_id: str
    market_question: str
    market_category: str | None
    side: str
    shares: Decimal
    avg_price: Decimal
    current_price: Decimal | None
    current_value: Decimal | None
    unrealized_pnl: Decimal | None
    cost_basis: Decimal
    status: str
    opened_at: datetime

    model_config = {"from_attributes": True}
